packages/core/src/connect.tsx

```diff
@@ -181,18 +179,16 @@ export const connect: Connect<any> = <P extends {}>(
       logger.debug("store", this.store);
 
       this.prevWatched = { ...this.watched };
-      this.firstTime = true;
       this.setState(this.watched);
       this.watched = {};
     }
 
     public shouldComponentUpdate(nextProps: P, nextState: any) {
       const test =
         !shallowEqual(this.props, nextProps) ||
-        (!this.firstTime && !shallowEqual(this.state, nextState));
```

Comment:
Why do we not need this anymore?

Reply:
I honestly never understood what it was for...

Reply:
In fact, I sort of do. Before, when we were calling synchronous init-type actions, they were being treated as async, so the component was rendered once with the old state, and then again when the action resolves. I believe this check was put in place to just ignore the first rerender (but it would only ever happen the first time, and not on subsequent rerenders).

This isn't necessary now, and in fact breaks things, since we'd be ignoring real refreshes. If the component is pure, comparing the state and props should be enough.